packages/checkout-ui-extensions/documentation/conditional-props.md

```diff
@@ -0,0 +1,88 @@
+- If button is focused, set the size to large
+- If button is hovered, set the size to large
+- If viewport's width is small, set the button's size to extraLarge
+- If viewport's width is small, and the focus is true, set the size to base
+
+### Style Helper class
+
+`Style` is a helper for authoring conditional values for prop styles.
+
+Write complex conditional styles based on one or more conditions (viewport sizes and interactive states) in a concise and expressive way.
+
+| Name    | Type                                                                                                                                                         | Description                                                                                                                                                                                                                                                                              |
+| ------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------ | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| default | <code>(defaultValue: T) => <a href="#chainableconditionalstyle">ChainableConditionalStyle</a><<wbr>T<wbr>></code>                                            | Sets an optional `defaultValue` to use when no other condition is met.                                                                                                                                                                                                                   |
+| when    | <code>(conditions: <a href="#conditions">Conditions</a>, value: T) => <a href="#chainableconditionalstyle">ChainableConditionalStyle</a><<wbr>T<wbr>></code> | Adjusts the style based on different `conditions`. All `conditions`, expressed as a literal object, must be met for the associated `value` to be applied.<br /><br />The `when` method can be chained together to build more complex styles, and are not necessarily mutually exclusive. |
+
+### Conditions
+
+The following conditions are supported for conditional styles.
+
+Multiple conditions can be set on the same `when` method.
+
+| Name                | Type                                                       | Description                                                                                     |
+| ------------------- | ---------------------------------------------------------- | ----------------------------------------------------------------------------------------------- |
+| hover?              | <code>true</code>                                          | Met when an element is hovered with the cursor (mouse pointer).                                 |
```

Comment:
`is hovered` might be missing a preposition ... previously you used `hovered on` for e.g. Maybe both should be `hovered over`?

Reply:
Agreed